### DISCUSSION TITLE
STT whisper not working when installed on local host machine

### DISCUSSION THREAD

**Discussion by JustARandomDude7** [open-webui/open-webui#4944] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.15
- **Ollama:** windows v0.3.6

- **Operating System:** Windows 11
- **Browser (if applicable):** Firefox

**Confirmation:**

- [V ] I have read and followed all the instructions provided in the README.md.
- [V ] I am on the latest version of both Open WebUI and Ollama.
- [V] I have included the browser console logs.
- [V] I have included the Docker container logs.
- [C] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

when pressing the STT button it should record your voice and turn it into text

## Actual Behavior:

Nothing Happens when trying to use the STT option. it logs my voice but when pressing the "V" button nothing happens. I have whisper on my local machine but it won't recognize it. I've tried also switching from whisper to the other options but stilll nothing

## Description

**Bug Summary:**
whisper API doesn't work for some reason

## Reproduction Details

**Steps to Reproduce:**
launch the docker container. open the webUI. select a model. press the record button. it shows a response to my voice (moving sound waves). pressing the confirm button ("V"). nothing happens

## Logs and Screenshots

**Browser Console Logs:**
Socket undefined disconnected due to transport close 0.DRpujwuF.js:1:80603
Additional details: 
Object { description: "websocket connection closed", context: close }
0.DRpujwuF.js:1:80662
Backend config: 
Object { status: true, name: "Open WebUI", version: "0.3.15", default_locale: "", oauth: {…}, features: {…}, default_models: null, default_prompt_suggestions: (7) […], audio: {…}, permissions: {…} }
+layout.svelte:88:11
user-count 
Object { count: 1 }
+layout.svelte:144:13
usage 
Object { models: [] }
+layout.svelte:149:13
connected +layout.svelte:123:13
Array(7) [ {…}, {…}, {…}, {…}, {…}, {…}, {…} ]
index.ts:64:9
Recording started VoiceRecording.svelte:165:11
Recording stopped VoiceRecording.svelte:171:11
TypeError: NetworkError when attempting to fetch resource. index.C9fEMm0Y.js:1:922
Socket undefined disconnected due to transport close +layout.svelte:135:13
Additional details: 
Object { description: "websocket connection closed", context: close }
+layout.svelte:137:14
Source map error: Error: NetworkError when attempting to fetch resource.
Resource URL: http://localhost:3000/_app/immutable/chunks/index.C9fEMm0Y.js
Source Map URL: index.C9fEMm0Y.js.map
user-count 
Object { count: 1 }
+layout.svelte:144:13
usage 
Object { models: [] }
+layout.svelte:149:13
connected +layout.svelte:123:13

**Docker Container Logs:**
2024-08-27 12:55:10 INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
2024-08-27 12:55:10 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
2024-08-27 12:55:10 INFO:     Application startup complete.
2024-08-27 12:55:10 INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
2024-08-27 12:55:14 INFO  [apps.openai.main] get_all_models()
2024-08-27 12:55:14 INFO  [apps.ollama.main] get_all_models()
2024-08-27 12:55:14 INFO:     172.17.0.1:45392 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P6KmrK3 HTTP/1.1" 200 OK
2024-08-27 12:55:14 INFO:     ('172.17.0.1', 45402) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=HZUgks2FQVnRlUkHAAAA" [accepted]
2024-08-27 12:55:14 INFO:     connection open
2024-08-27 12:55:14 user REDACTED
2024-08-27 12:55:14 INFO:     172.17.0.1:45392 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P6KmrKg&sid=HZUgks2FQVnRlUkHAAAA HTTP/1.1" 200 OK
2024-08-27 12:55:15 INFO:     172.17.0.1:45414 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P6KmrKi&sid=HZUgks2FQVnRlUkHAAAA HTTP/1.1" 200 OK

**Screenshots/Screen Recordings (if applicable):**
![Screenshot 2024-08-27 062926](https://github.com/user-attachments/assets/a0195a66-1c18-48c1-8635-a29c7fc75884)
![image](https://github.com/user-attachments/assets/36858cd0-538c-4eb7-8acf-4ff15ef94532)

## Additional Information
I like this webUI. very good work

**Comment by tjbck** [comment]
Can't seem to reproduce, let me know if you find the culprit for this issue!

**Reply by JustARandomDude7** [reply]
I've found the problem. I'm using the CUDA version of openwebui. I've switched to main and now it works. I think you should mention that the GPU support version (CUDA) is not supporting the STT option

**Reply by justinh-rahb** [reply]
Works on mine.

**Reply by KingstonJoseph12** [reply]
I'm facing the same issue now. Both main & cuda STT stopped working suddenly

**Comment by EgidioSuriano** [comment]
Hi I would also like to use Whisper to do audio to text transcription in Open WebUI. What is the procedure to install/install it? thanks

**Reply by mskry** [reply]
you have to install opendai-speech, but as mentioned by @JustARandomDude7 I have the same issue with cuda version on Open WebUI, but it works on a regular version (main)

`docker run -d --gpus=all -p 8000:8000 --add-host=host.docker.internal:host-gateway -v tts-voices:/app/voices -v tts-config:/app/config --name openedai-speech ghcr.io/matatonic/openedai-speech:latest`

**Comment by antonio-lamberti** [comment]
Same issue. IS there a solution for CUDA users?

**Comment by 4everfree** [comment]
I have faced the same issue.
